.travis.yml

```diff
@@ -29,59 +30,62 @@ env:
   global:
     - LUAROCKS=2.4.3
     - OPENSSL=1.0.2n
-    - CASSANDRA=3.9
     - OPENRESTY_BASE=1.13.6.2
     - OPENRESTY_LATEST=1.13.6.2
     - OPENRESTY=$OPENRESTY_BASE
     - DOWNLOAD_CACHE=$HOME/download-cache
     - INSTALL_CACHE=$HOME/install-cache
     - JOBS=2
+    - KONG_TEST_PG_USER=postgres
+    - KONG_TEST_PG_DATABASE=travis
+    - KONG_TEST_CASSANDRA_KEYSPACE=kong_tests
+    - KONG_TEST_DB_UPDATE_PROPAGATION=1
   matrix:
-    - TEST_SUITE=pdk
-    - TEST_SUITE=lint
-    - TEST_SUITE=unit
-    - TEST_SUITE=integration
-      OPENRESTY=$OPENRESTY_LATEST
-    - TEST_SUITE=plugins
-      OPENRESTY=$OPENRESTY_LATEST
-    - TEST_SUITE=integration
-      OPENRESTY=$OPENRESTY_BASE
-      CASSANDRA=2.2.12
-    - TEST_SUITE=plugins
-      OPENRESTY=$OPENRESTY_BASE
-      CASSANDRA=2.2.12
-    - TEST_SUITE=old-unit
-    - TEST_SUITE=old-integration
-      OPENRESTY=$OPENRESTY_BASE
-    - TEST_SUITE=old-plugins
-      OPENRESTY=$OPENRESTY_BASE
+    - KONG_TEST_DATABASE=postgres
+    - KONG_TEST_DATABASE=cassandra CASSANDRA=2.2
+    - KONG_TEST_DATABASE=cassandra CASSANDRA=3.9
 
 before_install:
   - source .ci/setup_env.sh
 
 install:
   - make dev
 
+cache:
+  apt: true
+  pip: true
+  directories:
+    - $DOWNLOAD_CACHE
+    - $INSTALL_CACHE
+    - $HOME/.ccm/repository
+
+stages:
+  - lint and unit
+  - pdk tests
+  - test
+  - name: deploy
+    if: tag =~ 0. || TRAVIS_EVENT_TYPE == cron
+
 jobs:
   include:
-    - stage: deploy
-      if: tag =~ 0.
-      script: .ci/trigger-travis.sh $TRAVIS_ACCESS_TOKEN
+    - stage: lint and unit
+      script:
+      - luacheck -q .
+      - bin/busted -v -o gtest spec/01-unit
+      - bin/busted -v -o gtest spec-old-api/01-unit
       env:
-        - TEST_SUITE=none
-    - stage: nightly deploy
-      if: TRAVIS_EVENT_TYPE == cron
+        - KONG_TEST_PG_DATABASE=foobar
```

Comment:
Why do we need this? Any particular unit test are relying on this not being set maybe? If so, could it be that this belongs to one of the setup scripts which could call `unset KONG_TEST_PG_DATABASE`? If not, then at the very least we could probably find a more elegant name for this value.

Reply:
It's the unit tests that are expecting `foobar`. We can move this to the `run_tests.sh` if we want I'm not sure of the benefits I always prefer things as ENV personally

Reply:
@hutchic Which test specifically? I don't think we ever needed to set this variable for the unit tests to succeed? :thinking:

Reply:
https://travis-ci.org/Kong/kong/jobs/413320585#L1794

```
[ RUN      ] spec/01-unit/003-prefix_handler_spec.lua @ 418: NGINX conf compiler prepare_prefix() dump Kong conf (custom conf)
spec/01-unit/003-prefix_handler_spec.lua:426: Expected objects to be the same.
```

Reply:
Something else must be broken with it, because this test sets its own value via the configuration loader arguments: https://github.com/Kong/kong/blob/master/spec/01-unit/003-prefix_handler_spec.lua#L420 It certainly shouldn't need a custom environment variable to be able to pass (that'd be a pretty terrible unit test...)